### PR TITLE
add 'buffer' type

### PR DESCRIPTION
**Pull request by juliangruber**
useful since this module can also be used from node. won't throw in the browser.

### CODE REVIEW

File: test/tests.js [hunk 1]
```diff
@@ -69,4 +69,12 @@ describe('type', function(){
   it('should match errors', function(){
     assert('error' === type(new Error('Ups!')));
   });
+
+  it('should match buffers', function(){
+    var b = {};
+    assert('object' === type(b));
+    window.Buffer = { isBuffer: function(val){ return val === b } };
```
Comment (dominicbarnes) [inline test/tests.js]: Can this test bypass this overwrite if `Buffer` already exists? (just as a sanity check when running via node)

Comment (juliangruber) [inline test/tests.js]: we didn't have a node test target in place so i didn't care about that

Comment (dominicbarnes) [inline test/tests.js]: hahaha, figures

**Comment by dominicbarnes** [comment]
:+1:

**Comment by juliangruber** [comment]
cool, thanks for the review dominic!

**Pull request merged**